pkg/controller/instance/plan_execution_engine_test.go

```diff
@@ -118,6 +118,37 @@ func TestExecutePlan(t *testing.T) {
 	}
 }
 
+func TestExecutePlanWithEnhancerError(t *testing.T) {
+	testClient := fake.NewFakeClientWithScheme(scheme.Scheme)
+	newStatus, err := executePlan(&activePlan{
+		name: "test",
+		PlanStatus: &v1alpha1.PlanStatus{
+			Status: v1alpha1.ExecutionPending,
+			Name:   "test",
+			Phases: []v1alpha1.PhaseStatus{{Name: "phase", Status: v1alpha1.ExecutionPending, Steps: []v1alpha1.StepStatus{{Status: v1alpha1.ExecutionPending, Name: "step"}}}},
+		},
+		spec: &v1alpha1.Plan{
+			Strategy: "serial",
+			Phases: []v1alpha1.Phase{
+				{Name: "phase", Strategy: "serial", Steps: []v1alpha1.Step{{Name: "step", Tasks: []string{"task"}}}},
+			},
+		},
+		tasks:     map[string]v1alpha1.TaskSpec{"task": {Resources: []string{"job"}}},
+		templates: map[string]string{"job": getResourceAsString(getJob("job1", "default"))},
+	}, &engineMetadata{
+		instanceName:        "Instance",
+		instanceNamespace:   "default",
+		operatorVersion:     "ov-1.0",
+		operatorName:        "operator",
+		resourcesOwner:      getJob("pod2", "default"),
+		operatorVersionName: "ovname",
+	}, testClient, &errKubernetesObjectEnhancer{})
+
+	if err == nil || newStatus == nil || newStatus.Status != v1alpha1.ErrorStatus || newStatus.Phases[0].Status != v1alpha1.ErrorStatus || newStatus.Phases[0].Steps[0].Status != v1alpha1.ErrorStatus {
+		t.Fatalf("Expecting to get an error with fatal error in status for error in using enhancer, got %w, %v", err, newStatus)
+	}
+}
```

Comment:
wdyt:
```suggestion
	assert. NoError(t, err)
	assert.Equal(t, newStatus.Phases[0].Status, v1alpha1.ErrorStatus, "Expected phase status to be an error")
	assert.Equal(t, newStatus.Phases[0].Step[0].Status, v1alpha1.ErrorStatus, "Expected step status to be an error")
```

Reply:
I was trying to use pure go instead of asserts but I actually don't care here that much so ... sure :) we're using assert everywhere anyway